src/components/SignInButton.js

```diff
@@ -1,24 +1,34 @@
+import { useEffect } from 'react'
 import { h } from 'react-hyperscript-helpers'
 import { Clickable } from 'src/components/common'
+import { spinner } from 'src/components/icons'
+import { isAuthSettled } from 'src/libs/auth'
+import { authStore } from 'src/libs/state'
 import * as Utils from 'src/libs/utils'
 
 
-const SignInButton = (props = {}) => {
-  Utils.useOnMount(() => {
-    window.gapi.signin2.render('signInButton', {
-      scope: 'openid profile email',
-      width: 250,
-      height: 56,
-      longtitle: true,
-      theme: 'light',
-      prompt: 'select_account'
-    })
-  })
+const SignInButton = ({ theme = 'light', ...props }) => {
+  const auth = Utils.useStore(authStore)
+
+  const isGoogleAuthInitialized = isAuthSettled(auth)
+
+  useEffect(() => {
+    if (isGoogleAuthInitialized) {
+      window.gapi.signin2.render('signInButton', {
+        scope: 'openid profile email',
+        width: 250,
+        height: 56,
+        longtitle: true,
+        theme,
+        prompt: 'select_account'
+      })
+    }
+  }, [isGoogleAuthInitialized, theme])
 
   // For some reason, Google's rendered Sign-In button is not at all keyboard accessible.
   // To fix this, we wrap it as a button, and propagate the keyboard-accessible click event down to
   // the inner DOM node inside the button, then let it bubble up to whatever it is that catches it.
-  return h(Clickable, {
+  return !isGoogleAuthInitialized ? h(spinner) : h(Clickable, {
```

Comment:
While this is in fact valid, the pattern in the codebase is to call `spinner()` directly; we should probably think about making it `h(Spinner)` at some point, but for now would you mind changing this for consistency?
```suggestion
  return !isGoogleAuthInitialized ? spinner() : h(Clickable, {
```